### DISCUSSION TITLE
Hardlink problem when balancing the pool

### DISCUSSION THREAD

**Discussion by racerlucas** [trapexit/mergerfs#1199] [Q&A]
When I tried to balance the pool by mergerfs.balance, something like this happended:
before:
disk 1: 90% full
/srv/disk1/.BT/Video/Movie
--movie1.mkv
--movie2.mkv
/srv/disk1/Video/Movie
--movie1.mkv(hardlinked, no actual space occupied)
--movie2.mkv(hardlinked, no actual space occupied)
disk 2: 80% full

after:
disk 1: 90% full
/srv/disk1/.BT/Video/Movie
--movie1.mkv
--movie2.mkv
disk 2: 90% full
/srv/disk2/Video/Movie
--movie1.mkv(actual space occupied)
--movie2.mkv(actual space occupied)

is that a bug? or I can fix that by changing some settings?

**Comment by trapexit** [comment]
No, it's not a bug. mergerfs.balance is just a simple tool. It's not intended for general use and does not do anything to attempt to preserve links. It literally scans the filesystem with the least % free space for the first file it finds and moves it to the filesystem with the most % free space. That's it. It was expressly designed for people to run once if they wanted `mfs` policy or similar and forgot to change from the default `epmfs`. If you want custom behaviors for moving stuff around you have to do it yourself. Everyone has there own expectations.
